src/commands/paranet/paranet-sync-command.js

```diff
@@ -450,35 +440,8 @@ class ParanetSyncCommand extends Command {
                         knowledgeAssetId,
                     );
 
-                // Create the .lock file indicating the data replication start
-                const paranetAssetLockPath =
-                    this.fileService.getParanetKnowledgeAssetLockDocumentPath(
-                        blockchain,
-                        contract,
-                        tokenId,
-                        blockchain,
-                        knowledgeAssetStorageContract,
-                        knowledgeAssetTokenId,
-                    );
-
-                let paranetAssetLockExists = await this.fileService.pathExists(
-                    paranetAssetLockPath,
-                );
-                if (!paranetAssetLockExists) {
-                    try {
-                        await this.fileService.writeContentsToFile(
-                            this.fileService.getParentDirectory(paranetAssetLockPath),
-                            `${blockchain}:${knowledgeAssetStorageContract}:${knowledgeAssetTokenId}.lock`,
-                            JSON.stringify({}),
-                            'wx',
-                        );
-                    } catch (error) {
-                        paranetAssetLockExists = true;
-                    }
-                }
-
                 let statePresentInParanetRepository = false;
-                if (!paranetAssetLockExists) {
+                if (!statePresentInParanetRepository) {
```

Comment:
I think here we need to check both if state is already present in repository + synced asset record

Reply:
Maybe, but those two should be same, if one is true the other one has to be true if everything is working corectly

Reply:
> Maybe, but those two should be same, if one is true the other one has to be true if everything is working corectly

It won't work in case when we have the same assertions for different KAs. Checking synced record should prevent this

Reply:
Why than not just check synced record?

Reply:
We can just check synced record, was just thinking to have double check to prevent issues when data wasn't inserted correctly